src/rgw/rgw_asio_client.cc

```diff
@@ -29,20 +25,21 @@ void ClientIO::init_env(CephContext *cct)
   env.init(cct);
 
   const auto& request = parser.get();
-  const auto& headers = request.fields;
+  const auto& headers = request;
   for (auto header = headers.begin(); header != headers.end(); ++header) {
-    const auto& name = header->name();
+    const auto& field = header->name(); // enum type for known headers
+    const auto& name = header->name_string();
     const auto& value = header->value();
 
-    if (boost::algorithm::iequals(name, "content-length")) {
+    if (field == beast::http::field::content_length) {
       env.set("CONTENT_LENGTH", value);
       continue;
     }
-    if (boost::algorithm::iequals(name, "content-type")) {
+    if (field == beast::http::field::content_type) {
       env.set("CONTENT_TYPE", value);
       continue;
     }
-    if (boost::algorithm::iequals(name, "connection")) {
+    if (field == beast::http::field::connection) {
       conn_keepalive = boost::algorithm::iequals(value, "keep-alive");
       conn_close = boost::algorithm::iequals(value, "close");
```

Comment:
oops..this isn't right. The Connection field is a comma-separated list of tokens. And the "keep-alive" property depends not only on the value in the field but also the HTTP version. Fortunately, `beast::http::message` provides a member function to do the dirty work for you:
http://www.boost.org/doc/libs/master/libs/beast/doc/html/beast/ref/boost__beast__http__message/keep_alive/overload1.html

If `keep_alive()` returns `true` then the connection has keep-alive semantics. Otherwise, it has `close` semantics. You only need to track one bool instead of two.